### DISCUSSION TITLE
What are C++’s best practices?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#106787] [Programming Help] [answered]
### Body

Hi,

My C++ code is always craptacular and because of that it usually has syntax errors or bugs and as a result my code is poorly designed and I am trying to fix that. Can any of you help?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by rishithamenush** [comment]
You can follow the below areas,

Learn C++ Core Guidelines.
Use modern C++ features.
Do code reviews and pair programming.
Learn refactoring and design patterns.
Use static analysis tools.
Practice with coding challenges.
Read books and take online courses.
Stay updated with the C++ community.

**Comment by ZephyrusBlaze** [comment, marked as answer]
Hey @PyDOS8,

Improving your C++ code can be done by following these best practices:
1. Use meaningful variable and function names.
2. Break down your code into smaller, manageable functions.
3. Comment your code to explain complex logic or algorithms.
4. Follow a consistent coding style and formatting.
5. Consider using a code formatter like [Prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) or [Clang Format](https://marketplace.visualstudio.com/items?itemName=xaver.clang-format) extensions for cleaner and consistent formatting.
